### ISSUE TITLE
Installing cadquery in a custom Dockerfile

### ISSUE THREAD

**Issue by hexad** [CadQuery/cadquery#708]
I use the cadquery editor in order to create shapes.
However, i would like the scripts i make, to run outside the editor's sandbox.

I have managed to import cadquery, when i performed the conda installation, using:
`conda create -n cq21-jl3 -c conda-forge -c cadquery python=3.8 cadquery`

However, i am still isolated in a conda environment, and my scripts don't run on the system's native environment.

So my next step was to look up the docker installation.

However, i do not want to have a docker image just for cadquery and jupyter notebook.

First of all, i am not interested in jupyter, i just want to run my scripts. In this case, the docker image in the official docs would work, but i am restricted only to cadquery for this image.

So what i would like, is to add cadquery to an existing dockerfile i use.

I searched online for the `bwalter42/jupyter_cadquery` image, to see if i could take a look at the dockefile, to add to my existing one. But i didn't have any luck.

Does anyone know how i can add cadquery to an existing dockerfile?

**Comment by jmwright** [comment]
@RubenRubens created a dockerfile [here](https://github.com/RubenRubens/cq-testing/blob/master/Dockerfile) that contains just CadQuery.

**Comment by hexad** [comment]
@jmwright Thank you. I see that he still uses it in a conda environment inside docker. So there's no escaping conda as far as i can tell, is this correct?

**Comment by jmwright** [comment]
> So there's no escaping conda as far as i can tell, is this correct?

Correct. OCP (our bindings to the OCCT CAD kernel) has a lot of dependencies and complex build/runtime environments. There is a desire on the part of the community to simplify installation, but Anaconda is required still.

**Comment by fedorkotov** [comment]
The only (easy) way to use CadQuery is conda (see #153). Also take look at mamba (#691) it is much faster.

**Comment by hexad** [comment]
Thank you guys. And could i ask for one more clarification. My dockerfile uses `FROM Ubuntu` as the base OS, while this dockerfile uses `FROM continuumio/miniconda3`.

I know the fact that i can have two `FROM` commands, but is step necessary?
Do you think i can have one `FROM ubuntu` and keep the rest the same as this dockerfile?

**Comment by jmwright** [comment]
@hexad I think the miniconda3 image only comes in Alpine and Debian variants. If you make sure Miniconda is installed, the same commands should work with your Ubuntu based docker file. I haven't tried it though.

**Comment by RubenRubens** [comment]
Here is an easy way to achieve it.
```
docker run --rm -v $PWD:/testing:z ghcr.io/rubenrubens/cq-testing:demo python cq_script.py
```

**Comment by RubenRubens** [comment]
If you are interested, consider checking out [this repo](https://github.com/RubenRubens/cq-containers)

**Comment by adam-urbanczyk** [comment]
Official images are here: https://github.com/orgs/CadQuery/packages
